dotnet/src/SemanticKernel.Abstractions/AI/Embeddings/IGenericEmbeddingGenerationService.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.SemanticKernel.Services;
+
+namespace Microsoft.SemanticKernel.Embeddings;
+
+/// <summary>
+/// Represents a generator of embeddings.
+/// </summary>
+/// <typeparam name="TValue">The type from which embeddings will be generated.</typeparam>
+/// <typeparam name="TEmbedding">The numeric type of the embedding data.</typeparam>
+[Experimental("SKEXP0001")]
+public interface IGenericEmbeddingGenerationService<TValue, TEmbedding> : IAIService
```

Comment:
I think `Generic` part in name is not needed here. `IEmbeddingGenerationService<TValue, TEmbedding>` should be good.

Reply:
@dmystrotruk look please at ADR PR. First I want to finish ADR.

Reply:
@dmytrostruk BTW. I can't catch you on discord.

Reply:
> @dmytrostruk look please at ADR PR. I am not sure which solution will be best so I created ADR to discuss.

@Krzysztof318 In this case it's worth to mark this PR as draft, so it will be clear that it's not ready for review yet :)

Reply:
done